src/ca/weblite/codename1/io/tar/TarEntry.java

```diff
@@ -138,7 +138,7 @@ public void setSize(long size) {
      */
     public boolean isDirectory() {
         if (this.file != null)
-            return FileSystemStorage.getInstance().isDirectory(this.file);
+            return FileSystemStorage.getInstance().isDirectory(FileSystemStorage.getInstance().getAppHomePath() + "/" + this.file);

```

Comment:
My gut tells me this isn't the best place to put this type of thing.  Can you open an issue and post a test case that demonstrates the problem?